### PR TITLE
test(maintenance): improve assertions on traces & logs

### PR DESCRIPTION
**Pull request by dreamorosi**
### Changes

> Please provide a summary of what's being changed

This PR improves and simplifies the helpers used to retrieve AWS X-Ray trace data that are used in the integration tests. As a result of this, the integration tests for the utility are also slightly simplified due to the helpers handling more of the data retrieval & validation from the X-Ray API.

The PR also extends some assertions in the Idempotency tests to more flexibly assert timeouts in Amazon CloudWatch Logs.

While working on this, I have also updated the Dependabot config so that it targets the `@aws-cdk/cli-lib-alpha` (dependency of `packages/testing`) as it had fallen behind on updates and was no longer compatible with the more recent AWS CDK installed version.

> Please add the issue number below, if no issue is present the PR might get blocked and not be reviewed

**Issue number:** #3011

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by dreamorosi** [comment]
Passing integration tests on this branch: https://github.com/aws-powertools/powertools-lambda-typescript/actions/runs/10671082450

File: packages/tracer/tests/helpers/invokeAllTests.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { invokeFunction } from '@aws-lambda-powertools/testing-utils';
+
+/**
+ * Invoke function sequentially 3 times with different parameters
+ *
+ * invocation: is just a tracking number (it has to start from 1)
+ * sdkV2: define if we will use `captureAWSClient()` or `captureAWS()` for SDK V2
```
Comment (am29d) [inline packages/tracer/tests/helpers/invokeAllTests.ts:7]:
```suggestion
```

Comment (am29d) [inline packages/tracer/tests/helpers/invokeAllTests.ts:7]:
```suggestion

```

Comment (am29d) [reply packages/tracer/tests/helpers/invokeAllTests.ts:7]: Remove unused param

File: packages/testing/src/types.ts [hunk 2]
```diff
@@ -87,6 +87,94 @@ interface TestStackProps {
...
+  startTime: Date;
+  resourceName: string;
+  expectedTracesCount: number;
+};
+
+type XRayTraceDocumentParsed = {
+  name: string;
+  id: string;
+  start_time: number;
+  end_time?: number;
```
Comment (am29d) [inline packages/testing/src/types.ts:102]: Why is in snake_case?

File: packages/idempotency/tests/e2e/makeHandlerIdempotent.test.ts [hunk 3]
```diff
@@ -263,9 +263,16 @@ describe('Idempotency E2E tests, middy middleware usage', () => {
...
-      // During the first invocation the function should timeout so the logs should contain 2 logs
-      expect(functionLogs[0]).toHaveLength(2);
-      expect(functionLogs[0][0]).toContain('Task timed out after');
+      try {
+        // During the first invocation the handler should be called, so the logs should contain 1 log
+        expect(functionLogs[0]).toHaveLength(2);
+        expect(functionLogs[0][0]).toContain('Task timed out after');
+      } catch {
+        // During the first invocation the function should timeout so the logs should not contain any log and the report log should contain a timeout message
+        expect(functionLogs[0]).toHaveLength(0);
```
Comment (am29d) [inline packages/idempotency/tests/e2e/makeHandlerIdempotent.test.ts:272]: I don't fully understand the catch here, can you elaborate?

**Review by am29d — APPROVED** [review]
Great work. The code duplication is a bit high for all the tracer tests (middy, decorator, manual), but I think we can address it in a different PR. We have many such cases where we need to check the same outcomes for various setups.

**Comment by dreamorosi** [comment]
> The code duplication is a bit high for all the tracer tests (middy, decorator, manual), but I think we can address it in a different PR

I agree. We have upcoming PRs where we'll work on streamlining similar to what we did in other areas in the past few weeks. For now seeing as the PR was already large I decided to not act on this.

**Pull request merged**